ts/components/IOMarkdown/types.ts

```diff
@@ -0,0 +1,50 @@
+import { VSpacer } from "@pagopa/io-app-design-system";
+import {
+  TxtHeaderNode,
+  TxtListNode,
+  TxtListItemNode,
+  TxtParagraphNode,
+  TxtStrNode,
+  ASTNodeTypes,
+  TxtStrongNode,
+  TxtEmphasisNode,
+  TxtLinkNode,
+  TxtImageNode,
+  TxtBlockQuoteNode,
+  TxtCodeBlockNode,
+  TxtCodeNode,
+  TxtBreakNode,
+  TxtHtmlNode,
+  TxtHorizontalRuleNode,
+  AnyTxtNode
+} from "@textlint/ast-node-types";
+import { ReactNode, ComponentProps } from "react";
+
+export type Renderer = (txtNode: AnyTxtNodeWithSpacer) => ReactNode;
+export type RuleRenderer<P, F extends Renderer | void = void> = (
+  param: P,
+  renderer: F
+) => ReactNode;
+export type AnyTxtNodeWithSpacer =
+  | AnyTxtNode
+  | { type: "Spacer"; size: number; key: string };
+
+export type IOMarkdownRenderRules = {
+  [ASTNodeTypes.Header]: RuleRenderer<TxtHeaderNode, Renderer>;
+  [ASTNodeTypes.Paragraph]: RuleRenderer<TxtParagraphNode, Renderer>;
+  [ASTNodeTypes.Str]: RuleRenderer<TxtStrNode, Renderer>;
+  [ASTNodeTypes.Strong]: RuleRenderer<TxtStrongNode, Renderer>;
+  [ASTNodeTypes.Emphasis]: RuleRenderer<TxtEmphasisNode, Renderer>;
+  [ASTNodeTypes.Comment]: RuleRenderer<null>;
+  [ASTNodeTypes.Link]: RuleRenderer<TxtLinkNode, Renderer>;
+  [ASTNodeTypes.List]: RuleRenderer<TxtListNode, Renderer>;
+  [ASTNodeTypes.ListItem]: RuleRenderer<TxtListItemNode, Renderer>;
+  [ASTNodeTypes.Image]: RuleRenderer<TxtImageNode, Renderer>;
+  [ASTNodeTypes.BlockQuote]: RuleRenderer<TxtBlockQuoteNode, Renderer>;
+  [ASTNodeTypes.CodeBlock]: RuleRenderer<TxtCodeBlockNode, Renderer>;
+  [ASTNodeTypes.Code]: RuleRenderer<TxtCodeNode, Renderer>;
+  [ASTNodeTypes.Break]: RuleRenderer<TxtBreakNode, Renderer>;
+  [ASTNodeTypes.Html]: RuleRenderer<TxtHtmlNode, Renderer>;
+  [ASTNodeTypes.HorizontalRule]: RuleRenderer<TxtHorizontalRuleNode, Renderer>;
+  Spacer: RuleRenderer<ComponentProps<typeof VSpacer> & { key: string }>;
+};
```

Comment:
```suggestion
type TxtNodeMap = {
  [ASTNodeTypes.Header]: TxtHeaderNode;
  [ASTNodeTypes.Paragraph]: TxtParagraphNode;
  [ASTNodeTypes.Str]: TxtStrNode;
  [ASTNodeTypes.Strong]: TxtStrongNode;
  [ASTNodeTypes.Emphasis]: TxtEmphasisNode;
  [ASTNodeTypes.Link]: TxtLinkNode;
  [ASTNodeTypes.List]: TxtListNode;
  [ASTNodeTypes.ListItem]: TxtListItemNode;
  [ASTNodeTypes.Image]: TxtImageNode;
  [ASTNodeTypes.BlockQuote]: TxtBlockQuoteNode;
  [ASTNodeTypes.CodeBlock]: TxtCodeBlockNode;
  [ASTNodeTypes.Code]: TxtCodeNode;
  [ASTNodeTypes.Break]: TxtBreakNode;
  [ASTNodeTypes.Html]: TxtHtmlNode;
  [ASTNodeTypes.HorizontalRule]: TxtHorizontalRuleNode;
};

type RendererMap = {
  [K in keyof TxtNodeMap]: RuleRenderer<TxtNodeMap[K], Renderer>;
} & {
  [ASTNodeTypes.Comment]: RuleRenderer<null>;
  Spacer: RuleRenderer<ComponentProps<typeof VSpacer> & { key: string }>;
};

export type IOMarkdownRenderRules = RendererMap;
};
```

What about this? Do you like it more? What do you think?

Reply:
Yes, I like it, thanks for this suggestion. I'll apply it!